pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -479,15 +479,8 @@ func buildGatewayNetworkFiltersFromTCPRoutes(node *model.Proxy, env *model.Envir
 		for _, tcp := range vsvc.Tcp {
 			if l4MultiMatch(tcp.Match, server, gatewaysForWorkload) {
 				upstream = tcp.Route[0].Destination // We pick first destination because TCP has no weighted cluster
-				destSvc, present := push.ServiceByHostname[model.Hostname(upstream.Host)]
```

Comment:
Does it mean that it's ok not to have a backend service?

Reply:
yes.. that was always the case. So, the "non.existent.cluster" will work (this is how we designed from teh start)